src/mon/OSDMonitor.cc

```diff
@@ -4120,17 +4120,23 @@ bool OSDMonitor::preprocess_command(MonOpRequestRef op)
 	  filter |= OSDMap::DUMP_IN;
 	} else if (s == "out") {
 	  filter |= OSDMap::DUMP_OUT;
+	} else if (s == "destroyed") {
+	  filter |= OSDMap::DUMP_DESTROYED;
 	} else {
 	  ss << "unrecognized state '" << s << "'";
 	  r = -EINVAL;
 	  goto reply;
 	}
       }
       if ((filter & (OSDMap::DUMP_IN|OSDMap::DUMP_OUT)) ==
-	  (OSDMap::DUMP_IN|OSDMap::DUMP_OUT) ||
-	  (filter & (OSDMap::DUMP_UP|OSDMap::DUMP_DOWN)) ==
-	  (OSDMap::DUMP_UP|OSDMap::DUMP_DOWN)) {
-	ss << "cannot specify both up and down or both in and out";
+	  (OSDMap::DUMP_IN|OSDMap::DUMP_OUT)) {
+        ss << "cannot specify both 'in' and 'out'";
+        r = -EINVAL;
+        goto reply;
+      }
+      if ((filter & (OSDMap::DUMP_UP|OSDMap::DUMP_DOWN|OSDMap::DUMP_DESTROYED)) ==
+	  (OSDMap::DUMP_UP|OSDMap::DUMP_DOWN|OSDMap::DUMP_DESTROYED)) {
+	ss << "can not specify 'up', 'down' and 'destroyed' simutaneously";
```

Comment:
This isn't quite right... we need to make sure they specify no more than one of the three and this allows any two.